### PR TITLE
Django testing - /posts url tests

### PR DESCRIPTION
**Pull request by nettapper**

### CODE REVIEW

**Comment by TianZhiWang** [comment]
Can you fix the conflicts first?

File: server/quickstart/tests/test_post.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        self.assertEqual(response.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    def test_put_405(self):
+        """ PUT should throw a client error as it doesn't make sense to put at this endpoint """
+        self.setUpAuthorLogin()
+        url = reverse("post")
+        response = self.client.put(url)
+        self.assertEqual(response.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    def test_get_only_returns_public(self):
```
Comment (kylecarlstrom) [inline server/quickstart/tests/test_post.py]: This is correct. We haven't update the post list to only return the public posts yet.

File: server/quickstart/tests/test_post.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        unAuthAuthor = Author.objects.create(displayName="Unauth", user=self.unAuthorizedUser)
+        unAuthAuthor.save()
+
+    def setUpAuthorLogin(self):
+        self.client.login(username='nixy', password='tester123')
+        self.client.force_authenticate(user=self.authorUser)
+
+    def setUpNotActiveLogin(self):
+        self.client.login(username="unauth", password="tester123")
+        self.client.force_authenticate(user=self.unAuthorizedUser)
```
Comment (kylecarlstrom) [inline server/quickstart/tests/test_post.py]: Yeah... this won't work. Calling force_authenticate bypasses authorization altogether. See here: http://www.django-rest-framework.org/api-guide/testing/#forcing-authentication

**Comment by nettapper** [comment]
@TianZhiWang if it wasn't already up to date, now it is
@kylecarlstrom good call, I'd seen that article before and until just now I couldn't get basic auth to work :/ BUT, I did figure it out :) so that problem has since been fixed!

**Pull request merged**